src/main/java/org/plumelib/bcelutil/StackMapUtils.java

```diff
@@ -630,6 +640,188 @@ protected final void update_uninitialized_NEW_offsets(InstructionList il) {
+
+      debug_instrument.log(
+          "gen_locals_from_byte_codes for offset: %d :: position: %d, inst: %s%n",
+          offset, ih.getPosition(), inst);
+
+      if (inst instanceof StoreInstruction) {
+        if (offset != ((LocalVariableInstruction) inst).getIndex()) {
+          continue;
+        }
+        stack = stack_types.get(ih.getPosition());
+        // get type of item about to be stored
+        Type tos = stack.peek(0);
+        // System.out.printf ("tos: %s, live_type: %s%n", tos, live_range_type);
+        // Store of a null does not change type.
+        // UNDONE: if tos is super of live_range_type, should not start new range
+        if (live_range_start == null || (!tos.equals(Type.NULL) && !tos.equals(live_range_type))) {
+          // close current live range
+          create_local_from_live_range(mgen, offset);
+          // start a new live range
+          live_range_type = tos;
+          live_range_start = ih.getNext();
+        }
+        // update live_range_end
+        live_range_end = ih.getNext();
+
+      } else if (inst instanceof IINC) {
+        if (offset != ((IndexedInstruction) inst).getIndex()) {
+          continue;
+        }
+        if (live_range_type == null) {
+          throw new RuntimeException("gen_locals_from_byte_code: no store before IINC");
+        } else if (live_range_type != Type.INT) {
+          throw new RuntimeException("gen_locals_from_byte_code: IINC operand not type int");
+        }
+        // update live_range_end
+        live_range_end = ih.getNext();
+
+      } else if (inst instanceof RET) {
+        if (offset != ((IndexedInstruction) inst).getIndex()) {
+          continue;
+        }
+        if (live_range_type == null) {
+          throw new RuntimeException("gen_locals_from_byte_code: no store before RET");
+        } else if (live_range_type.getType() != Const.T_ADDRESS) {
+          throw new RuntimeException(
+              "gen_locals_from_byte_code: RET operand not type returnAddress");
+        }
+        // update live_range_end
+        live_range_end = ih.getNext();
+
+      } else if (inst instanceof LoadInstruction) {
+        if (offset != ((LocalVariableInstruction) inst).getIndex()) {
+          continue;
+        }
+        stack = stack_types.get(ih.getPosition() + inst.getLength());
+        // get type of item about to be loaded
+        Type tos = stack.peek(0);
+        if (live_range_type == null) {
+          throw new RuntimeException("gen_locals_from_byte_code: no store before load");
+        } else if (!tos.equals(live_range_type)) {
+          // Load type can be super class of store type.  Rather than write code
+          // using reflection to verify, we just assume compiler got it right.
+          // throw new RuntimeException("gen_locals_from_byte_code: store/load types do not match");
+        }
+        // update live_range_end
+        live_range_end = ih.getNext();
+      }
+    }
+    if (live_range_end == null) {
```

Comment:
Each of the clauses above had a test for `(offset != inst.getIndex())` but this code at the end does not.  Is that intentional?

Reply:
Yes.  If we've reached the end of the method without seeing the end of the live range, we set it to be the end of the method.  Note that there may not be an active live_range but that will be checked in create_local_from_live_range.